src/Halogen/HTML/Events.purs

```diff
@@ -1,170 +1,129 @@
 -- | This module defines well-typed wrappers for common DOM events, so that
 -- | they may be safely embedded in HTML documents.
-
-module Halogen.HTML.Events
-  ( input
-  , input_
-  , onAbort
-  , onBeforeUnload
-  , onError
-  , onHashChange
-  , onLoad
-  , onPageShow
-  , onPageHide
-  , onResize
-  , onScroll
-  , onUnload
-  , onChange
-  , onInvalid
-  , onReset
-  , onSearch
-  , onSelect
-  , onSubmit
-  , onClick
-  , onContextMenu
-  , onDoubleClick
-  , onMouseDown
-  , onMouseEnter
-  , onMouseLeave
-  , onMouseMove
-  , onMouseOver
-  , onMouseOut
-  , onMouseUp
-  , onKeyDown
-  , onKeyPress
-  , onKeyUp
-  , onBlur
-  , onFocus
-  , onFocusIn
-  , onFocusOut
-  ) where
+module Halogen.HTML.Events where
 
 import Prelude
 
-import Data.Maybe
+import Control.Monad.Free (Free())
+
+import Data.Coyoneda (Coyoneda(), liftCoyoneda)
+import Data.Inject (Inject)
+
+import Halogen (actionF, actionFC)
+import Halogen.HTML.Events.Handler (EventHandler())
+import Halogen.HTML.Events.Types (Event(), MouseEvent(), FocusEvent(), KeyboardEvent())
+import Halogen.HTML.Core (Prop(), handler, eventName)
+
+inputF :: forall f g a. (Functor f, Functor g, Inject f g) => (forall i. a -> i -> f i) -> a -> EventHandler (Free g Unit)
```

Comment:
Are these all useful functions? Would be worthwhile explaining what they're used for.